feat(AI): Update Sentry AI docs (#11825)
getsentry/sentry-docs: docs/product/issues/issue-details/sentry-ai/index.mdx

---
title: Sentry AI
description: "Learn more about the AI features that help you understand and fix issues faster."
sidebar_order: 50
---

Sentry AI is a set of tools that help you quickly understand an issue, find a root cause, and create a fix. 
You can find it in the Solutions Hub in the sidebar of the issue details page for any error:

![Setup](./img/setup.png)

Once you've set up Sentry AI, you'll be able to use the following features:

- [**Issue Summary**](#issue-summary): A quick summary of all the data in the issue to help you understand what's going on at a glance.
- [**Autofix**](#autofix): A collaborative workflow to find the root cause of an error and create a PR that fixes it.

## Issue Summary

> 
> Issue Summary quickly orients you with the issue by summarizing key insights across event and issue level metadata.
> 
> When you open an issue, you'll see a quick overview of what's going wrong in the issue in the Solutions Hub in the sidebar:
> 
> ![Solutions Hub](./img/solutions-hub.png)
> 
> Opening the Solutions Hub will show you a more detailed overview of the issue. It will summarize what's going wrong, a possible cause, and if relevant, insights from trace-connected issues:
> 
> ![Issue Summary](./img/issue-summary.png)

Reviewer: ```suggestion

The Issue Summary provides a quick overview of an issue by highlighting key insights taken from event and issue level metadata. You'll see a quick overview of what's going wrong in the Solutions Hub in the sidebar:

![Solutions Hub](./img/solutions-hub.png)

When you open the Solutions Hub, you'll see a more detailed overview of the issue, including what's going wrong, a possible cause, and if relevant, insights from trace-connected issues.

![Issue Summary](./img/issue-summary.png)
```

## Autofix

> 
> _Note: Autofix replaces Suggested Fix, which was an earlier experiment that we have sunset._
> 
> When you want to understand the root cause more deeply, you can use Autofix. It uses the issue details, your codebase (integrated through GitHub), and its interactions with you to gain context and reason through the problem.
> 
> When you start an Autofix run, you'll see the following screen, with an option to share initial helpful context on the issue:
> 
> ![Autofix Start Screen](./img/autofix-start-screen.png)
> 
> Autofix then starts analyzing the issue and the relevant code. You can see its status at the bottom of the screen. As it goes through the process, Autofix will share its key insights along the way. You can expand each one to see the exact event data and code that it used to justify it. These insights build up a chain of reasoning to the root cause of the error.
> 
> Autofix won't know everything, so the best results come through collaboration. At any point, you can jump in with helpful context or feedback. You can also directly edit its line of reasoning. Just click the "Edit" button above any line, provide any thoughts or instructions you have, and Autofix will rethink from that point. Autofix can also recognize when it doesn't know something, and ask you for the context it needs to continue.
> 
> ![Autofix Insights](./img/autofix-qna.png)
> 
> Once you reach a root cause you're happy with, or have enough context to propose your own, Autofix will use the same process to find a fix. When complete, you'll see the diff of its proposed code changes. You can directly reject, edit, rethink, and give feedback on the changes. You can also ask Autofix to add tests to ensure the issue doesn't happen again.
> 
> ![Autofix Results](./img/autofix-results.png)
> 
> Once you're happy with the changes, Autofix can create a draft PR for you to review and merge in GitHub.

Reviewer: ```suggestion

<Note>

 Autofix replaces Suggested Fix (an earlier feature that has been sunset).
 
 </Note>

**Autofix** uses the issue details, your codebase (integrated through GitHub), and its interactions with you to gain context, reason through the problem, and get a deeper understanding of the root cause. 

When you start an Autofix run, you'll see the following screen, with an option to share any additional context:

![Autofix Start Screen](./img/autofix-start-screen.png)

Autofix will then analyze the issue and relevant code to help identify the root cause. You'll see the progress displayed at the bottom of the screen. Autofix will share key insights as they come up and you'll have the option to expand each. You'll see the event data and code that support Autofix conclusions, building a clear chain of reasoning back to the root cause.

Since Autofix doesn’t have all the context, collaboration leads to the best results. You can step in at any time to add context or feedback. To edit its reasoning, click the "Edit" button above any line to provide instructions or corrections, and Autofix will adjust its approach accordingly. If Autofix encounters gaps in its understanding, it will ask for your input.

![Autofix Insights](./img/autofix-qna.png)

When you’ve identified the root cause, or gathered enough information to propose your own solution, Autofix will use the same process to suggest a fix. It will present a diff of the proposed code changes. You can then:

- Accept, reject, or edit the changes.
- Provide feedback or ask Autofix to rethink its approach.
- Request tests to ensure the issue doesn’t recur.

![Autofix Results](./img/autofix-results.png)

Once you're happy with the changes, Autofix can create a draft PR for you to review and merge in GitHub.
```

## Data Processing

> We use the data mentioned below for analysis and to provide solutions for your review. Your data will not be used to train any models or to generate output shown to others. These features are powered by generative AI models hosted by the feature-specific subprocessors identified on our [subprocessor list](https://sentry.io/legal/subprocessors/). Our subprocessors will only use the data as directed by us.
> 
> The data used for these features includes:
> 
> - Error messages
> - Stack traces
> - Sentry spans
> - DOM interactions
> - Relevant code from linked repositories
> 

Reviewer: ```suggestion
We use the data listed below for analysis and to provide solutions for your review. Your data will not be used to train any models or to generate output shown to others. These features are powered by generative AI models hosted by the feature-specific subprocessors identified on our [subprocessor list](https://sentry.io/legal/subprocessors/). Our subprocessors will only use the data as directed by us.

The data used for these features includes:

- Error messages
- Stack traces
- Sentry spans
- DOM interactions
- Relevant code from linked repositories

```

---
title: Sentry AI

> description: "Learn more about AI that helps you understand and fix issues faster."

Reviewer 2: ```suggestion
description: "Learn more about the AI features that help you understand and fix issues faster."
```

sidebar_order: 50
---

Sentry AI is a set of tools that help you quickly understand an issue, find a root cause, and create a fix. 
You can find it in the Solutions Hub in the sidebar of the issue details page for any error:

![Setup](./img/setup.png)

Once you've set up Sentry AI, you'll be able to use the following features:

- [**Issue Summary**](#issue-summary): A quick summary of all the data in the issue to help you understand what's going on at a glance.

> - [**Autofix**](#autofix): An agent that collaborates with you to find the root cause of an error and create a PR that fixes it.

Reviewer 2: ```suggestion
- [**Autofix**](#autofix): A collaborative workflow to find the root cause of an error and create a PR that fixes it.
```

## Issue Summary

> Issue Summary quickly orients you with the issue by summarizing the key insights to know across all the issue details.

Reviewer 2: ```suggestion
Issue Summary quickly orients you with the issue by summarizing key insights across event and issue level metadata.
```

When you open an issue, you'll see a quick overview of what's going wrong in the issue in the Solutions Hub in the sidebar:

![Solutions Hub](./img/solutions-hub.png)

Opening the Solutions Hub will show you a more detailed overview of the issue. It will summarize what's going wrong, a possible cause, and if relevant, insights from trace-connected issues:

![Issue Summary](./img/issue-summary.png)

## Autofix

When you want to understand the root cause more deeply, you can use Autofix. It uses the issue details, your codebase (integrated through GitHub), and its interactions with you to gain context and reason through the problem.

When you start an Autofix run, you'll see the following screen, with an option to share initial helpful context on the issue:

![Autofix Start Screen](./img/autofix-start-screen.png)

> Autofix then starts analyzing the issue and the relevant code. You can see it's status at the bottom of the screen. As it goes through the process, Autofix will share its key insights along the way. You can expand each one to see the exact event data and code that it used to justify it. These insights build up a chain of reasoning to the root cause of the error.

Reviewer 2: ```suggestion
Autofix then starts analyzing the issue and the relevant code. You can see its status at the bottom of the screen. As it goes through the process, Autofix will share its key insights along the way. You can expand each one to see the exact event data and code that it used to justify it. These insights build up a chain of reasoning to the root cause of the error.
```

> Autofix obviously doesn't know everything, so it collaborates with you. At any point, you can jump in with helpful context or feedback. You can also directly edit its line of reasoning. Just click the "Edit" button above any of them, provide any thoughts or instructions you have, and Autofix will walk back its thought process and rethink from that point. Autofix can also recognize when it doesn't know something, and ask you for the context it needs to continue.

Reviewer 2: ```suggestion
Autofix won't know everything, so the best results come through collaboration. At any point, you can jump in with helpful context or feedback. You can also directly edit its line of reasoning. Just click the "Edit" button above any line, provide any thoughts or instructions you have, and Autofix will rethink from that point. Autofix can also recognize when it doesn't know something, and ask you for the context it needs to continue.
```

![Autofix Insights](./img/autofix-qna.png)

Once you reach a root cause you're happy with, or have enough context to propose your own, Autofix will use the same process to find a fix. When complete, you'll see the diff of its proposed code changes. You can directly reject, edit, rethink, and give feedback on the changes. You can also ask Autofix to add tests to ensure the issue doesn't happen again.

![Autofix Results](./img/autofix-results.png)

Once you're happy with the changes, Autofix can create a draft PR for you to review and merge in GitHub.

## Data Processing

> Under normal circumstances we do not send data to a third party
> undisclosed in our [subprocessor list](https://sentry.io/legal/subprocessors/).
> Because of this you need to provide consent to do so. If your organization has
> signed a DPA with us, an owner of the organization will also need to sign a
> "OpenAI Subprocessor Acknowledgement" in the organization [Legal & Compliance
> page](https://sentry.io/orgredirect/organizations/:orgslug/settings/legal/). Otherwise an individual
> user will receive a prompt to confirm that data will be sent to a third party.
> 
> The data shared via this feature will not be used to train any AI models. The data used for these features includes:

Reviewer 2: ```suggestion
We use the data mentioned below for analysis and to provide solutions for your review. Your data will not be used to train any models or to generate output shown to others. These features are powered by generative AI models hosted by the feature-specific subprocessors identified on our [subprocessor list](https://sentry.io/legal/subprocessors/). Our subprocessors will only use the data as directed by us.

The data used for these features includes:
```

- Error messages
- Stack traces
- Sentry spans
- DOM interactions
- Relevant code from linked repositories

> 

Reviewer 2: ```suggestion

You can learn more about our data privacy practices [here](https://docs.sentry.io/security-legal-pii/security/ai-ml-policy/#use-of-identifying-data-for-generative-ai-features).
```
